### PR TITLE
Add ability to label samples with metadata (e.g. Pangolin designation)

### PR DESCRIPTION
**Pull request by hyanwong**

### CODE REVIEW

**Comment by hyanwong** [comment]
Screenshot 2024-09-25 at 18 05 32

**Comment by hyanwong** [comment]
Actually, country would also be useful here, right?

File: sc2ts/info.py [hunk 1]
```diff
@@ -1440,7 +1440,18 @@ class SampleGroupInfo:
...
+            self,
+            size=(800, 600),
+            time_scale=None,
+            y_axis=True,
+            mutation_labels=None,
+            style=None,
+            highlight_universal_mutations=None,
+            x_regions=None,
+            node_labels=None,
+            node_label_key=None,
```
Comment (jeromekelleher) [inline sc2ts/info.py:1453]: This is a nice idea, but it requires that the thing you want to add is in the metadata, and that the metadata is something you want to display "as is". How about we do something more direct like say
```
if node_labels == "strain":
    node_labels = {u: md["strain"] for ...
elif node_labels == "pango":
    ...
elif node_labels == "country":
   ...
```
That way we can tune it to make it easy to show the stuff we're using? This is just a debugging tool, so we can just add what we need quickly and not worry about long term stuff.

**Review by jeromekelleher — COMMENTED** [review]
It's nice, but see counter-proposal?

File: sc2ts/info.py (hunk 1 above)
Comment (hyanwong) [reply sc2ts/info.py:1453]: Yep, that would work. I thought labelling by Pango + (abbreviated) country would be most useful, and we could easily have a "pango+country" option.

Comment (jeromekelleher) [reply sc2ts/info.py:1453]: sure

Comment (hyanwong) [reply sc2ts/info.py]: Done. Should be able to merge.

**Pull request merged**